guillotina/transactions.py

```diff
@@ -131,7 +131,7 @@ async def __aexit__(self, exc_type, exc, tb):
                 self.txn.deleted = {**self.previous_txn.deleted, **self.txn.deleted}
                 self.txn.added = {**self.previous_txn.added, **self.txn.added}
 
-        if self.abort_when_done:
+        if self.abort_when_done or exc:
```

Comment:
also, i guess we don't want to run execute_futures on exception, right?

Reply:
Yes, definitely!